src/components/Articles.jsx

```diff
@@ -1,26 +1,35 @@
 import { useEffect, useContext, useState } from "react"
-import { fetchArticles } from "../../axios.js"
+import { fetchArticles, fetchArticlesByTopic } from "../../axios.js"
 import ArticleCard from "./ArticleCard.jsx"
 import { ArticleContext } from "../contexts/ArticleContext.jsx"
+import SearchByTopic from "./SearchByTopic.jsx"
 
-export default function Articles (props) {
+export default function Articles ({ articles, setArticles, topic_slug }) {
 
-    const { articles, setArticles } = props
     const { setArticle } = useContext(ArticleContext)
     const [isLoading, setIsLoading] = useState(true)
     const [isErr, setIsErr] = useState(false)
 
     useEffect(() => {
-        fetchArticles().then(({data}) => {
-            setIsLoading(false)
-            setArticles(data.articles)
-        })
-        .catch((err) => {
-            setIsLoading(false)
-            setIsErr(true)
-            console.log(err)
-        })
-    },[])
+        topic_slug === "All" || !topic_slug? 
```

Comment:
please refactor all this functionality to your fetchArcticles function and remove this conditional fetching of articles

Reply:
Hi, could you expand on this please, I'm confused about what I'm being asked to do, thanks.